vm.c

```diff
@@ -29,6 +29,32 @@
     pc = n;        \
     goto *labels[OPCODE.opcode]
 
+#define VM_PUSH(n)                           \
+    do {                                     \
+        sp--;                                \
+        env->temps[sp].type = INT;           \
+        env->temps[sp].value.vint = (int) n; \
+    } while (0)
+
+#define VM_POP(n)                               \
+    do {                                        \
+        n = (size_t) env->temps[sp].value.vint; \
+        sp++;                                   \
+    } while (0)
+
+#define VM_CALL(n)   \
+    do {             \
+        VM_PUSH(pc); \
+        VM_GOTO(n);  \
+    } while (0)
+
+#define VM_RET()    \
+    do {            \
+        from = pc;  \
```

Comment:
I prefer not to use macro. Instead, specialized inline functions or even code generators are generally better to me.

Please check [what v8 does](https://github.com/v8/v8/blob/e38011326cb710e432598df14cfc7d8f9497ee5a/src/assembler.cc#L389).